src/video_core/rasterizer_interface.h

```diff
@@ -49,15 +49,32 @@ class RasterizerInterface {
     /// Records a GPU query and caches it
     virtual void Query(GPUVAddr gpu_addr, QueryType type, std::optional<u64> timestamp) = 0;
 
+    /// Signal a GPU based semaphore as a fence
+    virtual void SignalSemaphore(GPUVAddr addr, u32 value) = 0;
+
+    /// Signal a GPU based syncpoint as a fence
+    virtual void SignalSyncPoint(u32 value) = 0;
+
+    /// Release all pending fences.
+    virtual void ReleaseFences() = 0;
+
     /// Notify rasterizer that all caches should be flushed to Switch memory
     virtual void FlushAll() = 0;
 
     /// Notify rasterizer that any caches of the specified region should be flushed to Switch memory
     virtual void FlushRegion(VAddr addr, u64 size) = 0;
 
+    virtual bool MustFlushRegion(VAddr addr, u64 size) = 0;
```

Comment:
```suggestion
    virtual bool MustFlushRegion(VAddr addr, u64 size) const = 0;
```

This member function also needs a documentation comment.

Reply:
Can't ber const as the caches can't be const due to their respective mutex.

Reply:
You can mark the mutex as mutable.